.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,158 @@
+name: PR
+
+on:
+  pull_request
+
+jobs:
+  python-dependencies:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
+          echo "::set-env name=PYTHON_VERSION::$(cat .python-version)"
+      - uses: actions/setup-python@v1
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+      - name: Install apt dependencies
+        run: sudo apt-get install libsnappy-dev libgconf-2-4
+      - name: Install Pipenv
+        run: pip install pipenv==2018.11.26
+      - name: Cache virtualenv
+        id: cache-virtualenv
+        uses: actions/cache@v1
```

Comment:
Actions should be referenced by commit hash rather than tag.
```yaml
uses: actions/cache@a505c2e
```

Tags can be opaquely updated, so this ensures that the expected code is checked out. See https://julienrenaux.fr/2019/12/20/github-actions-security-risk/ for further explanation

Reply:
These aren't random actions in github marketplace, they are ones maintained in the official github org.

Reply:
True - I was aiming more to set a precedent from the start so that should any future developers add actions from marketplace the sha is used. I'm not sure where else is best to capture that information :man_shrugging:

Reply:
I think the other thing is that this applies to other dependencies of ours - e.g. pulling images from docker. 

Those tags could theoretically be updated. We also assume the latest versions of packages should be used in pip through using pipenv, so there could be risks there we may not catch too.

I think until there is something that can maintain commit hashes for us within actions, I suggest we lock to sha's for non-github marketplace actions?

Reply:
I agree - use version for github actions and sha's for non-github actions.